lib/seattleflu/db/cli/command/etl/manifest.py

```diff
@@ -88,48 +90,170 @@ def etl_manifest(*, action: str):
+                    utm_collection_barcode = manifest_record.document.pop("utm", None)
+                    utm_sample_id = None
+                    utm_status = "None"
+                    
+                    if utm_collection_barcode:
+
+                        utm_collection_identifier = check_barcode(db, 
+                            record          = manifest_record,
+                            barcode         = utm_collection_barcode,
+                            identifier_set  = expected_identifier_sets["kit_collections"])
+
+                        # Skip record if no matching UTM collection identifier
+                        # found for the UTM collection barcode    
+                        if not utm_collection_identifier: continue
+                        
+                        # There should only be a UTM sample barcode if there
+                        # is a UTM collection barcode. 
+                        utm_sample_barcode = manifest_record.document.pop("utm_sample", None)
+
+                        # Convert UTM sample barcode to full identifier and check
+                        # it's from the correct identifier set.
+                        if utm_sample_barcode:
+                            utm_sample_identifier = check_barcode(db, 
+                                record          = manifest_record,
+                                barcode         = utm_sample_barcode,
+                                identifier_set  = expected_identifier_sets["samples"])
+                            
+                            # Skip record if no matching UTM sample identifier
+                            # found for the UTM sample barcode
+                            if not utm_sample_identifier: continue
+
+                        else:
+                            # If the UTM does not have a sample barcode, then
+                            # just assign the UTM collection identifier as the
+                            # UTM sample identifier.
+                            utm_sample_identifier = utm_collection_identifier
+
+                        # Pop off UTM sample details from manifest record
+                        utm_details = utm_sample_details(manifest_record.document)
+
+                        # Upsert the UTM sample
+                        utm_sample, utm_status = upsert_sample(db,
+                            identifier            = utm_sample_identifier.uuid,
+                            collection_identifier = utm_collection_identifier.uuid,
+                            additional_details    = utm_details)
+
+                        utm_sample_id = utm_sample.id
+
+                    # Convert RDT collection barcode to full identifier and
+                    # check it's from the correct identifier set.
+                    rdt_collection_barcode = manifest_record.document.pop("rdt", None)
+                    rdt_sample_id = None
+                    rdt_status = "None"
```

Comment:
Why is `rdt_status` a string here?

Reply:
All statuses are strings so I thought it would be more specific for it to say "None" rather than have a null value. But that can be changed if a null value is enough.

Reply:
I think a null-value would be more appropriate since nulls are the special value intended for representing "unknown" or "missing", which is what we have here. Adding the string-literal `"None"` as one of those special values will require more special-casing in querying code later.